sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/KeyVaultCredentialPolicy.java

```diff
@@ -1,125 +1,182 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
-
 package com.azure.security.keyvault.administration.implementation;
 
 import com.azure.core.credential.TokenCredential;
 import com.azure.core.credential.TokenRequestContext;
 import com.azure.core.http.HttpPipelineCallContext;
-import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpRequest;
 import com.azure.core.http.HttpResponse;
-import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.http.policy.BearerTokenAuthenticationPolicy;
 import com.azure.core.util.CoreUtils;
-import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.Locale;
 import java.util.Map;
-import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 
 /**
- * A policy that authenticates requests with Azure Key Vault service. The content added by this policy is leveraged
- * in {@link TokenCredential} to get and set the correct "Authorization" header value.
+ * A policy that authenticates requests with the Azure Key Vault service. The content added by this policy is
+ * leveraged in {@link TokenCredential} to get and set the correct "Authorization" header value.
  *
  * @see TokenCredential
  */
-public final class KeyVaultCredentialPolicy implements HttpPipelinePolicy {
-    private final ClientLogger logger = new ClientLogger(KeyVaultCredentialPolicy.class);
-    private static final String WWW_AUTHENTICATE = "WWW-Authenticate";
+public class KeyVaultCredentialPolicy extends BearerTokenAuthenticationPolicy {
     private static final String BEARER_TOKEN_PREFIX = "Bearer ";
-    private static final String AUTHORIZATION = "Authorization";
-    private final ScopeTokenCache cache;
+    private static final String CONTENT_LENGTH_HEADER = "Content-Length";
+    private static final String KEY_VAULT_STASHED_CONTENT_KEY = "KeyVaultBody";
+    private static final String KEY_VAULT_STASHED_CONTENT_LENGTH_KEY = "KeyVaultBodyContentLength";
```

Comment:
We may want to think about making these even more unique as accidentally overwriting them will break a request